src/uct/ib/ud/base/ud_ep.c

```diff
@@ -954,14 +960,14 @@ void uct_ud_ep_process_rx(uct_ud_iface_t *iface, uct_ud_neth_t *neth, unsigned b
         /* must be connection request packet */
         uct_ud_ep_rx_creq(iface, neth);
         goto out;
-    } else if (ucs_unlikely(!ucs_ptr_array_lookup(&iface->eps, dest_id, ep) ||
-                            (ep->ep_id != dest_id)))
```

Comment:
this is not needed really, because if we found it in ptr_array it means dest_id is same?
maybe assert it?